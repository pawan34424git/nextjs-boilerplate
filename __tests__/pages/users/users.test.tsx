import "@testing-library/jest-dom";
import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import {
  mockDefaultGetUsersFail,
  mockDefaultGetUsersSuccess,
} from "../../../__mocks__/users/users.mock";
import UsersPage from "../../../pages/users/index";

describe("User List page", () => {
  afterEach(cleanup);

  it("user edit rendering", () => {
    const tree = renderer.create(<UsersPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("page table render", async () => {
    mockDefaultGetUsersSuccess();
    render(<UsersPage />);

    const heading = screen.getByText("User List");
    expect(heading).toBeInTheDocument();

    await waitFor(() => screen.getByTestId("table"));

    const tableElement = screen.getByTestId("table");
    expect(tableElement).toBeInTheDocument();

    const noDataText = screen.queryByText(/no data found/);
    expect(noDataText).not.toBeInTheDocument();

    const editButton = screen.queryAllByText(/Edit/);
    fireEvent.click(editButton[0]);
  });

  it("page table render fails", async () => {
    mockDefaultGetUsersFail();
    render(<UsersPage />);

    const noDataText = await screen.findByText(/no data found/);
    expect(noDataText).toBeInTheDocument();
  });
});
