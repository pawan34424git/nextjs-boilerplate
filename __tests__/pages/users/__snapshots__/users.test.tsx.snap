// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User List page user edit rendering 1`] = `
<div
  className="min-h-full"
>
  <main>
    <div
      className="flex-col rounded m-0 p-0 dark:bg-slate-800  shadow-inner px-2 py-1 "
    >
      <div
        className="min-w-full"
      >
        <h1>
          User List
        </h1>
        <p>
          no data found!
        </p>
      </div>
    </div>
  </main>
</div>
`;
